omod_1.x/src/main/java/org/openmrs/module/openhmis/inventory/web/controller/JasperReportController.java

```diff
@@ -288,7 +288,11 @@ private String renderReport(int reportId, HashMap<String, Object> parameters, St
 		}
 
 		try {
-			ReportGenerator.generate(report, parameters, false, true);
+			if (Context.getAuthenticatedUser() == null) {

```

Comment:
This check should be done as the first part of the render method so that no other processing is done. Performing the check here means that all the work to parse the request and get the report must be done only to ignore it all and redirect to the login page.